CMake target exports are not created if not using system curl

### Description

When building and installing cpr using curl that is not installed in the system no CMake export sets are created, possibly leading to errors in projects dependent on cpr. When installing projects dependent on cpr the following CMake error might pop up:

```CMake Error: install(EXPORT "myProject_Targets" ...) includes target "myProject" which requires target "cpr" that is not in any export set.```

Is there a specific reason why the exports are omitted in this case? 

For most builds this is not an issue, but it can become an issue if CMakes `FetchContent` together with find_package is used to build a dependency tree. (This is not a very good way to handle dependencies, but it beats checking in binaries to git)

### Example/How to Reproduce

See the CMakeLists.txt in the following example. 
https://github.com/bernedom/CPR_Export_Failure_Example 

To build it use 
```
cmake -S . -B ./build
```

Configuration should fail with the following error: 
`CMake Error: install(EXPORT "unleash_Targets" ...) includes target "unleash" which requires target "cpr" that is not in any export set.`

### Possible Fix

the fix would be changing ln 81ff in In the file https://github.com/libcpr/cpr/blob/master/cpr/CMakeLists.txt like this:
```
 install(TARGETS cpr
                EXPORT cprTargets
                RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
                ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
                LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
                
install(EXPORT cprTargets
                FILE cprTargets.cmake
                NAMESPACE cpr::
                DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/cpr) 
```

### Where did you get it from?

GitHub (branch e.g. master)

### Additional Context/Your Environment

CMake version 3.25 or higher (possibly other versions are also affected)

--- a/cpr/CMakeLists.txt
+++ b/cpr/CMakeLists.txt
@@ -49,13 +49,13 @@ set_target_properties(cpr
 # Import GNU common install directory variables
 include(GNUInstallDirs)
 
-if(CPR_USE_SYSTEM_CURL)
-        install(TARGETS cpr
-                EXPORT cprTargets
-                RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-                ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
-                LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS cpr
+        EXPORT cprTargets
+        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
+if(CPR_USE_SYSTEM_CURL)
         # Include CMake helpers for package config files
         # Follow this installation guideline: https://cmake.org/cmake/help/latest/manual/cmake-packages.7.html
         include(CMakePackageConfigHelpers)
@@ -69,17 +69,11 @@ if(CPR_USE_SYSTEM_CURL)
                 "${PROJECT_BINARY_DIR}/cpr/cprConfig.cmake"
                 INSTALL_DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/cpr)
 
-        install(EXPORT cprTargets
-                FILE cprTargets.cmake
-                NAMESPACE cpr::
-                DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/cpr)
-
         install(FILES ${PROJECT_BINARY_DIR}/cpr/cprConfig.cmake
                 ${PROJECT_BINARY_DIR}/cpr/cprConfigVersion.cmake DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/cpr)
-
-else()
-        install(TARGETS cpr
-                RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-                ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
-                LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
 endif()
+
+install(EXPORT cprTargets
+        FILE cprTargets.cmake
+        NAMESPACE cpr::
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/cpr)

